pkg/event/processor.go

```diff
@@ -146,12 +146,14 @@ func NewBatchEventProcessor(options ...BPOptionConfig) *BatchEventProcessor {
 }
 
 // Start initializes the event processor
-func (p *BatchEventProcessor) Start(exeCtx utils.ExecutionCtx) {
+func (p *BatchEventProcessor) Start(ctx context.Context) {
 	if p.EventDispatcher == nil {
-		p.EventDispatcher = NewQueueEventDispatcher(exeCtx.GetContext())
+		dispatcher := NewQueueEventDispatcher()
+		defer dispatcher.flushEvents()
+		p.EventDispatcher = dispatcher
 	}
 
-	p.startTicker(exeCtx)
+	p.startTicker(ctx)
 	pLogger.Debug("Batch event processor started")
```

Comment:
With these changes, does this message actually only get logged when the processor is stopped? Since `startTicker` was changed to return only when the context signals done.

Reply:
Good catch, I'll move the log line.